app/services/courses/creation_service.rb

```diff
@@ -26,6 +26,12 @@ def build_new_course
       course = provider.courses.new
       course.assign_attributes(course_attributes.except(:subjects_ids, :study_mode))
 
+      if course_attributes[:master_subject_id].blank? && course_attributes[:subordinate_subject_id].present?
+        course.errors.add(:subjects, :course_creation)
+      elsif subject_ids.present? || course_attributes[:level] == 'further_education'
+        AssignSubjectsService.call(course:, subject_ids:)
+      end
+
```

Comment:
This is here so we can avoid validating the subjects unless under specific circumstances.

We don't want to call `AssignSubjectsService.call(course:, subject_ids:)` when we have not reached the subjects form (choosing primary / secondary level)
We don't want to call it if there are no subject ids either, except in the case when the level is `further education` because no subject ids are passed for this level type and the Service is responsible for assigning the FurtherEducation subject to that course.